src/providers/profile/profile.ts

```diff
@@ -1065,6 +1073,21 @@ export class ProfileProvider {
           }
 
           this.profile = Profile.fromObj(profile);
+
+          if (credentials && credentials.token) {
+            const tokenWallet = this.profile.credentials.find(
+              oldCredentials =>
+                oldCredentials.coin == credentials.token.symbol.toLowerCase() &&
+                oldCredentials.walletId == credentials.walletId
+            );
+            if (!tokenWallet) {
+              this.logger.info(`Adding Token ${credentials.token.symbol}`);
+              this.profile.credentials.push(credentials);
+              this.profile.dirty = true;
+              this.storeProfileIfDirty();
+            }
+          }
+
```

Comment:
This is the key method to create ERC20 token wallets. It duplicates ETH wallet credentials, and saves it to `profile.credentials` localStorage.